amazon/iontest/ion_test_driver.py

```diff
@@ -741,13 +749,397 @@ def parse_implementations(descriptions, output_root):
             for description in descriptions]
 
 
+def write_errors(report, first_field, first_report, second_field, second_report, field, msg):
+    report[TestFile.ERROR_MESSAGE_FIELD] = msg
+    errors = {}
+    if first_field is not None:
+        errors[first_field] = first_report
+    if second_field is not None:
+        errors[second_field] = second_report
+    report[field] = errors
+
+
+def write_to_report(cur_result, final_result, check_report, file, field):
+    cur_result[field] = check_report
+    final_result[file] = cur_result
+
+
+def get_name(location):
+    name = location.split("/")[-1]
+    if '.' in name:
+        name = name[0:name.rfind('.')]
+    return name
+
+
+def get_name_for_write(location):
+    location_array = location.split("/")
+    # implementation's name
+    impl = location_array[-4]
```

Comment:
Sure, in case people modify the result file manually. 

The path after `results/` are created by ion-test-driver. I added code to check if the file structure are valid for children under the dir `'results'`.